lib/indico/settings.rb

```diff
@@ -0,0 +1,94 @@
+require 'inifile'
+
+module Indico
+  private
+
+  class << self; attr_accessor :config; end
+
+  def self.valid_auth(config)
+    # Does a config hashmap have a valid auth definition?
+    return config['auth'] &&
+           config['auth']['api_key']
+  end
+
+  def self.valid_cloud(config)
+    # Does a config hashmap have a valid private cloud definition?
+    return config['private_cloud'] &&
+           config['private_cloud']['cloud']
+  end
+
+  def self.merge_config(prev_config, new_config)
+    # Merge two configurations, giving precedence to the second
+    if self.valid_auth(new_config)
+      prev_config['auth'] = new_config['auth']
+    end
+
+    if self.valid_cloud(new_config)
+      prev_config['private_cloud'] = new_config['private_cloud']
+    end
+
+    return prev_config
+  end
+
+  def self.load_config_files(filenames)
+    # Given a list of filenames, build up an ini file config
+    master_ini = self.new_config()
+
+    for filename in filenames
+      ini = IniFile.load(filename)
+      if not ini then
+        next
+      end
+      master_ini = self.merge_config(master_ini, ini)
+    end
+    return master_ini
+  end
+
+  def self.find_config_files()
+    # Provides a list of paths to search for indicorc files
+    localPath = File.join(Dir.pwd, ".indicorc")
+    globalPath = File.join(Dir.home, ".indicorc")
+    return [globalPath, localPath]
+  end
+
+  def self.load_environment_vars()
```

Comment:
I really feel like there should be a better way to write this, but don't have the ruby chops to know how it should be done -- any edits here are much appreciated, because three calls to `Hash.new` feels a bit wrong to me.